test/index.js

```diff
@@ -5,16 +5,21 @@
 import assert from 'node:assert/strict'
 import fs from 'node:fs/promises'
 import process from 'node:process'
-import test from 'node:test'
-import highlight from 'highlight.js'
+import test, {beforeEach, mock} from 'node:test'
+import highlight from 'highlight.js/lib/core'
 import {rehype} from 'rehype'
 import {removePosition} from 'unist-util-remove-position'
-import {lowlight} from '../index.js'
+import {lowlight} from '../lib/all.js'
```

Comment:
Why mock? Wouldn’t it be better to actually test whether the two interfere?

Reply:
As the configuration can't be read, we would have to compare the results of `highlight` and `lowlight`. Would you prefer that instead of checking if `configure` is called?

Reply:
Btw. I had to switch to `../lib/all.js`, as not all languages of the `fixture` folder are covered by the `common` setup.

And another topic - locally I saw this failing check, but the message wasn't clear to me 😬 

```
/home/runner/work/lowlight/lowlight/lib/all.d.ts:2:13: DoNotTochItRegistersLanguageFiles
1846 / 1847 99.94%
The type coverage rate(99.94%) is lower than the target(100%).
```

Reply:
To get the tests working, I think this is the trick:

```diff
 import highlight from 'highlight.js'
+import coffeescript from 'highlight.js/lib/languages/coffeescript'
+import haskell from 'highlight.js/lib/languages/haskell'
+import http from 'highlight.js/lib/languages/http'
+import pgsql from 'highlight.js/lib/languages/pgsql'
```

```diff
+lowlight.registerLanguage('coffee', coffeescript)
+lowlight.registerLanguage('haskell', haskell)
+lowlight.registerLanguage('http', http)
+lowlight.registerLanguage('pgsql', pgsql)
```

I also think you’d also need:

```diff
-  assert.deepEqual(
+  assert.notDeepEqual(
     lowlight.listLanguages(),
     expectedLanguages,
-    'should return the same list of languages as highlight.js'
+    'should *not* return the same list of languages as highlight.js'
   )
 
```

The tests failing for this actually shows that everything works, so that’s great :)

Reply:
With this, I actually think it’s fine to not add highlight.js tests. Or mocking: so these changes you made to the tests.

But I am fine with an actual test, that checks that if a language is registered with lowlight, it still crashes when trying to use it with regular highlight.js?

Reply:
As for the `all.d.ts` problem: it happens because you now use `all.js`. It isn’t really a problem. Without `all.js` it’s fine. Or you can add it next to `common.d.ts` in `package.json`: https://github.com/wooorm/lowlight/blob/5caa24559b7ff858e3009f10545ee19ef9c64d00/package.json#L98C6-L98C17

Reply:
> To get the tests working, I think this is the trick:

👍 I switched back to the `../index.js` and added the missing languages like you proposed.

---

> I also think you’d also need:
>
> ```
> -  assert.deepEqual(
>+  assert.notDeepEqual(
>    lowlight.listLanguages(),
>      expectedLanguages,
> -    'should return the same list of languages as highlight.js'
> +    'should *not* return the same list of languages as highlight.js'
>   )
> ```

To check that case I added the check if the manually registered `testtest` is not available within `highlight.js` - see [here](https://github.com/wooorm/lowlight/pull/51/files#diff-5bb8db779819ddef5956a5d9d5949c05ef7445237656ca37bf2f02720271440bR328-R331). That's a bit more clear than just comparing the lists, as e.g. highlight.js might change their own set of languages. 

The reason why I stumbled over this issue was, that the `configure` has been called globally. That's the main reason why I wanted to make a test explicit for that case. Just to explain why I added that 😉 but yeah, it's implicitly tested within the other tests 👍

Reply:
@wooorm what do you think about the state of this PR? Should I make any further changes?

Reply:
It looks good! Thing for me is that this is a breaking change. And at this point there's no need to stick with the current hljs-like global API. Meaning I need to think about how this should look.
And I'm AFK a bit these weeks!

Reply:
> And at this point there's no need to stick with the current hljs-like global API.

Curious why you felt that was a requirement before?

Reply:
> Meaning I need to think about how this should look.
> And I'm AFK a bit these weeks!

Thanks for the response @wooorm. Let me know if I can help out somehow ✌️

Reply:
@joshgoebel because this was created years ago, when hljs supported much less, no emitters for example. And global. So it was a subclass. Now that can change. Perhaps a similar API to https://github.com/wooorm/starry-night. The `flagToScope` part particularly might be nice to have, which is a bit involved, I’d need some mapping between GH linguist and HLJS grammars

Reply:
> Meaning I need to think about how this should look. And I'm AFK a bit these weeks!

@wooorm did you already have time to think about the options? Let me know if I can support somehow ✌️

Reply:
Haven’t found the time yet, I’m busy with major updates through the ±500 projects maintained in the unified collective.
That being said, in a couple days I’ll reach https://github.com/rehypejs/rehype-highlight, which uses this, at which point I’d like to have this change in it. So I’ll come up with something.